packages/sdk/src/light_push/light_push.ts

```diff
@@ -86,40 +106,94 @@ export class LightPush implements ILightPush {
       pubsubTopic: encoder.pubsubTopic
     });
 
-    const coreResults: CoreProtocolResult[] =
+    // Track protocol versions used per peer
+    const protocolVersions: Record<string, string> = {};
+
+    const coreResults: LightPushCoreResult[] =
       peerIds?.length > 0
         ? await Promise.all(
-            peerIds.map((peerId) =>
-              this.protocol.send(encoder, message, peerId).catch((_e) => ({
-                success: null,
-                failure: {
-                  error: ProtocolError.GENERIC_FAIL
+            peerIds.map(async (peerId) => {
+              try {
+                const result = await this.protocol.send(
+                  encoder,
+                  message,
+                  peerId
+                );
+
+                // Enhanced error logging with protocol version information
+                if (result.failure) {
+                  const peerIdStr = peerId.toString();
+                  const protocolVersion =
+                    result.failure.protocolVersion ||
+                    (result.failure.statusCode !== undefined ? "v3" : "v2");
+                  protocolVersions[peerIdStr] = protocolVersion;
+
+                  log.warn(
+                    `Failed to send to peer ${peerIdStr} (${protocolVersion}): ${result.failure.error}`,
+                    {
```

Comment:
result of this function must not change 
consumers do not need status codes etc

we have discussed it already

Reply:
need to change return type because of https://github.com/waku-org/js-waku/issues/2409